machine-learning/_posts/2018-10-18-UsefulThingsML.md

```diff
@@ -0,0 +1,193 @@
+
+The main role of theoretical guarantees in machine learning is not as a
+criterion for practical decisions, but as a source of understanding and
+driving force for algorithm desig
+
+Just because a learner has a theoretical justification and works in practice
+practice doesn't mean the former is the reason for the latter.
+
+
+7. **FEATURE ENGINEERING IS THE KEY**
+
+At the end of the day, some machine learning projects succeed and some fail.
+What makes the difference? Easily the most important factor is the features
+used. **Often, the raw data is not in a form that is amenable to learning, but
+you can construct features from it that are**.
+
+Machine learning is not a one-shot process of building a data set and running
+a learner, but rather an iterative process of running the learner, analyzing
+the results, modifying the data and/or the learner, and repeating.
+
+**Features that look irrelevant in isolation may be relevant in combination**.
+
+
+8. **MORE DATA BEATS A CLEVERER ALGORITHM**
+
+Pragmatically the quickest path to success is often to just get more data. As
+a rule of thumb, a dumb algorithm with lots and lots of data beats a clever
+one with modest amounts of it.
+
+Enormous mountains of data are available, but there is not enough time to
+process it, so it goes unused. Even though in principle more data means that
+more complex classifiers can be learned, **in practice simpler classifiers end
+up being used, because complex ones take too long to learn**. **Part of the
+answer is to come up with fast ways to learn complex classifiers**.
+
+**Part of the reason using cleverer algorithms has a smaller payoff than you
+might expect is that, to a first approximation, they all do the same**.
+
+**Because of the curse of dimensionality**, no existing amount of data may be
+enough. For these reasons, **clever algorithms often pay off in the end**.
+
+**In the end, the biggest bottleneck is not data or CPU cycles, but human
+cycles**.
+
+
+9. **LEARN MANY MODELS, NOT JUST ONE**
+
+The best learner varies from application to application. If instead of
+selecting the best variation found, we combine many variations, the results
+are better --often much better-- and at little extra effort for the user.
+
+Model ensembles examples: bagging, boosting,etc.
+
+
+10. **SIMPLICITY DOES NOT IMPLY ACCURACY**
+
+Occam's razor: entities should not be multiplied beyond necessity. In machine
+learning, this is often taken to mean that, **given two classifiers with the
+same training error, the simpler of the two will likely have the lowest test
+error**.
+
+**But** counter-examples exist: there is no necessary connection between the
+number of parameters of a model and its tendency to overfit (e.g. SVMs).
+
+The conclusion is that simpler hypotheses should be preferred because
+simplicity is a virtue in its own right, not because of a hypothetical
+connection with accuracy.
+
+
+11. **REPRESENTABLE DOES NOT IMPLY LEARNABLE**

```

Comment:
Not sure what's the takeaway of this section.

Reply:
I'd say that they mean that not every model is useful in any context. Re-worded in 9bd2c4c .